cruditor/views.py

```diff
@@ -272,29 +277,6 @@ def format_linked_objects(self, objects):
             for obj in objects
         ]
 
-    def form_valid(self, form):
-        """
-        Call ``perform_delete`` method and show a nice success message. If there
-        are protected related objects, an error message is shown instead with
-        the output of ``format_linked_objects``.
-        """
-        try:
-            self.perform_delete(form)
-        except models.ProtectedError as e:
-            return self.render_to_response(self.get_context_data(
-                linked_objects=self.format_linked_objects(e.protected_objects)))
-        messages.success(self.request, self.success_message.format(
-            model=self.get_model_verbose_name(), object=self.object))
-        return super().form_valid(form)
-
-    def get_context_data(self, **kwargs):
-        """
-        Set ``form_save_button_label`` context variable to change the button
```

Comment:
I'd like to keep the option to rename the confirmation button. Can we restore this?

Reply:
Yes, for sure, I have missed this option. I have restored this the same way it is done in `templates/cruditor/form.html` - i.e. the default label is defined in the template. I also add an extra test... Is it okay for you?